### ISSUE TITLE
MA0027 is only reported when Exception is caught

### ISSUE THREAD

**Issue by drieseng** [meziantou/Meziantou.Analyzer#1025]
### Version of the Meziantou.Analyzer NuGet package

3.0.11

### Rule Identifier

MA0027

### Target Framework

.NET 10.0

### C# Language version

C# 14

### Description

**MA0027** appears to only be reported if the caught exception is exact of the exact type **System.Exception** and not for a derived type.

### Reproduction Steps

Minimal code:

```c#
try
{
}
catch (InvalidOperationException ex)
{
    throw ex; // Only CA2200 and S3445 are reported
}
```

### Other information

I see the following options:
* Fix MA0027
* Deprecate it in favor of CA2200
